include/boost/beast/websocket/impl/read.hpp

```diff
@@ -563,7 +563,7 @@ class stream<NextLayer, deflateSupported>::read_some_op
                     else if(impl.rd_fh.fin)
                     {
                         // append the empty block codes
-                        std::uint8_t constexpr
+                        static std::uint8_t constexpr
```

Comment:
I was wondering if `static` is needed. It _shouldn't_ be. But who knows. I am not very proficient with constexpr.

Reply:
Same here, the only problem I was able to find later was since this is created inside the `else if` scope, the stack will unwind the the `sz.next_in` could potentially point to overwritten bytes, our release version with optimizations turned on was crashing with `zlib::invalid_block_length`.

Reply:
Whether constexpr or const results in the same code: 
https://godbolt.org/z/1j9z18

It definitely has to be static otherwise it's dangling pointer.
In this case, gcc actually optimises out the initialisation of empty_block.
https://godbolt.org/z/9xd3jv